roles/self_hosted_additional_host/tasks/main.yml

```diff
@@ -11,16 +11,6 @@
   tags:
     - packages
 
-- name: get the provisioning nic for the machine
-  shell: |
-     ip addr | grep -Pzo '^[0-9]+:(.*):.*\n.*{{ mac_address }}.*' | sed -r -n 's/[0-9]+: (.*):.*/\1/p' | head -n1
```

Comment:
ipv4.interface is the interface with the IP and I guess that's the NIC you want for provisioning.